File: src/Blog/Blog.Blazor/Data/Entities/ResourceTranslation.cs
using Blog.Blazor.Data;

namespace Blog.Blazor.Data.Entities;

public sealed class ResourceTranslation
{
    public int Id { get; set; }
    public int ResourceId { get; set; }
    public string Locale { get; set; } = string.Empty;
    public string Title { get; set; } = string.Empty;
    public string? AlternativeTitle { get; set; }
    public string? Description { get; set; }
    public string? PersonalNote { get; set; }
    public string? ReasonFound { get; set; }

    [JsonOrEmpty]
    public string? Seo { get; set; }

    public DateTime? CreatedAt { get; set; }
    public DateTime? UpdatedAt { get; set; }

    public Resource? Resource { get; set; }
}

File: laravel/app/Models/ResourceTranslation.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class ResourceTranslation extends Model
{
    use HasFactory;

    protected $fillable = ['resource_id', 'locale', 'title', 'alternative_title', 'description', 'personal_note', 'reason_found', 'seo'];

    protected $casts = ['seo' => 'array'];

    /**
     * @return BelongsTo<resource, $this>
     */
    public function resource(): BelongsTo
    {
        return $this->belongsTo(Resource::class);
    }
}
